tests/daemon/test-daemon.c

```diff
@@ -678,10 +678,11 @@ get_events_from_request (GByteArray    *request,
 
   gint64 client_relative_time, client_absolute_time;
   gchar *image_version;
+  GVariant *site_id;
```

Comment:
Try to have the child first:
```
+  gsize num_siteid_values = g_variant_n_children (site_id);
+  g_info ("%s: site_id has %ld children", __func__, num_siteid_values);
+  for (gsize i = 0; i < num_siteid_values; i++)
+    {
+      gchar *key, *val;
+      g_variant_get_child(site_id, i, "{ss}", &key, &val);
+      g_info ("%s: {%s, %s}", __func__, key, val);
+    }
+  g_variant_unref (site_id);
```
The weird things are:
* **site_id** has no children!???
*  **site_id** cannot be unreferenced.  It shows error `Bail out! GLib-FATAL-CRITICAL: g_atomic_ref_count_dec: assertion 'g_atomic_int_get (arc) > 0' failed`.

Reply:
Also, checked `create_request_body()`'s [**site_id**](https://github.com/endlessm/eos-event-recorder-daemon/commit/de50c7bac8abba911c9a6ccb787e660bab1c28a3#diff-fadc9fc0899639577ac993bdc978725b109a6ec299404b9eb2817eb3df59c972R851) in `daemon/emer-daemon.c`.

It does get correct content:`[{"id", "myid"}, {"country", "Earth"}]` from [`emer_read_location_label()`](https://github.com/endlessm/eos-event-recorder-daemon/commit/de50c7bac8abba911c9a6ccb787e660bab1c28a3#diff-b9a781d12fdd6a794c150bb50aade43c8c6a9288ed50a8c2286e6e910ee3a6baR27) invoked by [`emer_get_site_id()`](https://github.com/endlessm/eos-event-recorder-daemon/commit/de50c7bac8abba911c9a6ccb787e660bab1c28a3#diff-b9a781d12fdd6a794c150bb50aade43c8c6a9288ed50a8c2286e6e910ee3a6baR40) in `tests/daemon/mock-site-id-provider.c`.

Reply:
GVariant always confuses me. The issue is that `site_id` is referring to a dictionary `a{ss}`. In that case, per the [dictionaries documentation](https://developer.gnome.org/glib/stable/gvariant-format-strings.html#gvariant-format-strings-dictionaries), the format passed to `g_variant_get` should be `@a{ss}` to refer to a pointer to a `GVariant`. Then per the [GVariant *](https://developer.gnome.org/glib/stable/gvariant-format-strings.html#gvariant-format-strings-gvariant) documentation, this will set the pointer to a reference, so (I believe) it should be dereferenced when done.

The following worked for me:
```
diff --git a/tests/daemon/test-daemon.c b/tests/daemon/test-daemon.c
index dce5e30..73d10d9 100644
--- a/tests/daemon/test-daemon.c
+++ b/tests/daemon/test-daemon.c
@@ -655,11 +655,23 @@ get_events_from_request (GByteArray    *request,
   GVariant *site_id;
   guchar boot_type;
   g_variant_get (native_endian_request,
-                 "(xxsa{ss}ya(aysxmv)a(ayssxmv))",
+                 "(xxs@a{ss}ya(aysxmv)a(ayssxmv))",
                  &client_relative_time, &client_absolute_time, &image_version,
                 &site_id, &boot_type, singular_iterator, aggregate_iterator);
   g_free (image_version);
 
+  const gchar *site_value;
+  gboolean site_ret;
+  g_assert_nonnull (site_id);
+  g_assert_cmpint (g_variant_n_children (site_id), ==, 2);
+  site_ret = g_variant_lookup (site_id, "id", "&s", &site_value);
+  g_assert_true (site_ret);
+  g_assert_cmpstr (site_value, ==, "myid");
+  site_ret = g_variant_lookup (site_id, "country", "&s", &site_value);
+  g_assert_true (site_ret);
+  g_assert_cmpstr (site_value, ==, "Earth");
+  g_variant_unref (site_id);
+
   g_assert_cmpint (client_relative_time, >=, fixture->relative_time);
   g_assert_cmpint (client_relative_time, <=, curr_relative_time);
 
```

Reply:
Uh!  The '@' character again!  It always confuses me.  Thanks @dbnicholson